### PR TITLE
update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by chensuyue**
## Type of Change

documentation 

## Description

Update CONTRIBUTING.md.   
1. More details for PR create, request from #1303 
2. Add Pull Request Acceptance Criteria
3. Add Pull Request status checks overview

## Expected Behavior & Potential Risk

Better help contributions

## How has this PR been tested?

Spell check pass

## Dependency Change?

No

### CODE REVIEW

**Review by XuehaoSun — APPROVED** [review]

File: docs/source/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,43 +1,57 @@
...
-**Dependency Change?**  
-any library dependency introduced or removed
+## Pull Request Status Checks Overview
+Intel® Neural Compressor use [Azure DevOps](https://learn.microsoft.com/en-us/azure/devops/pipelines/?view=azure-devops) for CI test.
+And generally use [Azure Cloud Instance](https://azure.microsoft.com/en-us/pricing/purchase-options/pay-as-you-go) to deploy pipelines, e.g. Standard E16s v5.
+|     Test Name                 |     Test Scope                                |     Test Pass Criteria    |
+|-------------------------------|-----------------------------------------------|---------------------------|
+|     Code Scan                 |     Pylint/Bandit/CopyRight/DocStyle/SpellCheck       |     PASS          |
+|     [DCO](https://github.com/apps/dco/)     |     Use `git commit -s` to sign off     |     PASS          |
+|     Unit Test                 |     Pytest scripts under [test](/test)                |      PASS (No failure, No core dump, No segmentation fault, No coverage drop)      |
```
Comment (thuang6) [inline docs/source/CONTRIBUTING.md:53]: Do we have test coverage target or current coverage number available to contributor if he/she want to check if there is a drop?  or the UT result will report current coverage and new coverage with PR?

Comment (chensuyue) [reply docs/source/CONTRIBUTING.md:53]: The UT test will report a comparison table between this PR and master, so user could check the coverage and also able to see with file has regression. It's hard for user to get a full coverage data of their PR locally, it require several test ENV and combine all the test results to get the full coverage.

Comment (chensuyue) [reply docs/source/CONTRIBUTING.md:53]: Like this one: 
image

Comment (thuang6) [reply docs/source/CONTRIBUTING.md:53]: I just experienced similar UT coverage result in my PR to ITREX, thanks for the clarification

**Review by thuang6 — APPROVED** [review]

**Pull request merged**